components/eventing-controller/pkg/subscriptionmanager/jetstream/jetstream_test.go

```diff
@@ -0,0 +1,154 @@
+package jetstream
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	kymalogger "github.com/kyma-project/kyma/common/logging/logger"
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	"github.com/kyma-project/kyma/components/eventing-controller/logger"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/eventtype"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/subscriptionmanager"
+	controllertesting "github.com/kyma-project/kyma/components/eventing-controller/testing"
+	"github.com/nats-io/nats.go"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type jetStreamSubMgrMock struct {
+	Client  dynamic.Interface
+	Backend handlers.JetStreamBackend
+}
+
+func (c *jetStreamSubMgrMock) Init(_ manager.Manager) error {
+	return nil
+}
+
+func (c *jetStreamSubMgrMock) Start(_ env.DefaultSubscriptionConfig, _ subscriptionmanager.Params) error {
+	return nil
+}
+
+func (c *jetStreamSubMgrMock) Stop(_ bool) error {
+	return nil
+}
+
+func TestCleanup(t *testing.T) {
+	jsSubMgr := jetStreamSubMgrMock{}
```

Comment:
Is `jsSubMgr` required for this test ? From what I see it only servers as a container for `Client` and `Backend`.

Reply:
And it's just mocking the behaviour and calls of the actual `SubscriptionManager`. Honestly, I don't think its needed either. We can discuss this.